lib/storage/sqlite/campaign-info-storage.js

```diff
@@ -53,4 +53,22 @@ class CampaignInfoStorage {
   }
 }
 
+/**
+ * Fetch a CampaignInfoByCandidate object by id
+ *
+ * @param {Number} id candidate_id
+ * @param {campaignInfoProcessor} cb
+ */
+CampaignInfoStorage.prototype.fetchByCampaign = function (id, cb) {
+  db.open(this.dbpath)
+    .then((stmt) => stmt.all('SELECT * FROM campaign_info WHERE candidate_id = ?', id))
+    .then((rows) => {
+      rows.forEach((row) => {
+        return new CampaignInfo(row);
```

Comment:
I don't think that returning within this callback actually does anything. If you're trying to create an array of instantiated `CampaignInfo` object from the array of rows, maybe [`Array.prototype.map`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map) is a better choice here?